docs/README.md

```diff
@@ -6,3 +6,21 @@
  * [KFserving data plane specification](data-plane.md)
    * Defines the data payloads to enable interoperability between `kfserving` components.
 
+# Architecture Overview
+The KFService Data Plane architecture consists of a static graph of components. The diagram may not be up to date with the current set of features available in the graph, but the overall mechanism holds:
+
+- The User defines the components they wish to use in a KFService.
+  - The only required component is a Predictor.
+  - Additional components may be specified to attach addtional behavior. 
+  - Each component is specified at the ModelSpec layer, allowing for canarying.
+- The KFService will support one of several Data Plane interfaces.
+  - For example: cloud-ai-platform/HTTP, cloud-ai-platform/GRPC, seldon/HTTP, seldon/GRPC
```

Comment:
Need more clarity on that. This means if you have backend models "deployed" on Seldon or GCP, you can use this in front to route et all? Or you can even "deploy" and manage models on Seldon and GCP through this?

Reply:
This is not a frontend. This is merely talking about the "Data Plane" interfaces. This is all running on the users cluster. Can you explain more about what's confusing and I can try to address the language.

Reply:
I think we need to front-end it by explaining what do we mean by CONTROL and DATA Plane in the context of Model Serving. To me there are three parts in user journey
1. Deploy the model
2. Use model for prediction
3. Manage the model (canary et all)

In which part do I need to know about  Data Plane interfaces supported..?

Reply:
Ok so it should be #2  for the above case. Now then, going back to the implementation part, we will use TF serving data plane interface for XGBoost and TF, or if someone want to use Seldon as the backend engine here, then they will getting response according to Seldon`s schema?

Reply:
See lines 1-7:

# Design Docs
 * [KFserving control plane specification](control-plane.md)
   * Defines the specification of the `kfserving` resource.
 * [KFserving data plane specification](data-plane.md)
   * Defines the data payloads to enable interoperability between `kfserving` components.

1 & 3 are control plane, 2 is data plane

Reply:
> To me KFserving should become a black box, and should "tell" me one way to use data plane, with schema defined. So we pick one between TFServing or Seldon, try to make it standard here

I want to get here eventually, but I think the reality of merging existing serving systems creates challenges for this. Changing control planes is always easier than changing data planes. We don't, for example, want to impede the performance of tensorflowserving by putting a translation layer in front of it.

I'm working with Clive and Robbie to standardize these data plane definitions, but this is a longer effort that will require careful thought and things like upstreaming features to TensorflowServing. 

Short term, Clive and I agree we'll need to support n-small Data Plane APIs
Long term, we will see if making a standardized & efficient master protocol is viable, and then deprecate the others.